test/integration/targets/apt/tasks/apt.yml

```diff
@@ -437,3 +437,48 @@
   file:
     path: /usr/sbin/policy-rc.d
     state: absent
+
+# https://github.com/ansible/ansible/issues/65325
+- name: install hello with apt (to test allow_change_held_packages option)
+  apt:
+    name: hello
+    state: present
+    allow_change_held_packages: yes
+  register: apt_result
+
+- name: check hello with dpkg
+  shell: dpkg-query -l hello
+  failed_when: False
+  register: dpkg_result
+
+- name: verify installation of hello
+  assert:
+    that:
+      - "apt_result.changed"
+      - "dpkg_result.rc == 0"
+
+- name: verify apt module outputs
+  assert:
+    that:
+      - "'changed' in apt_result"
+
+# INSTALL AGAIN
+- name: install hello with apt
+  apt:
+    name: hello
+    state: present
+    allow_change_held_packages: yes
+  register: apt_result
+
+- name: verify no change on re-install
+  assert:
+    that:
+      - "not apt_result.changed"
+
+# UNINSTALL
+- name: uninstall hello with apt
+  apt:
+    name: hello
+    state: absent
+    purge: yes
+  register: apt_result
```

Comment:
The cleanup should be done as a handler. `ansible-tests` runs with `ANSIBLE_FORCE_HANDLERS=True`, so handlers always run even if other tasks fail.

Reply:
Ok did not see that. I will change it.